source/Halibut/Transport/Protocol/MessageSerializer.cs

```diff
@@ -62,9 +73,13 @@ public T ReadMessage<T>(Stream stream)
         T ReadCompressedMessage<T>(Stream stream)
         {
             using (var zip = new DeflateStream(stream, CompressionMode.Decompress, true))
-            using (var bson = new BsonDataReader(zip) { CloseInput = false })
+            using (var decompressedObservableStream = new ObservableStream(zip))
```

Comment:
Would it make sense to observe the zip side as well, since we may decide to have the zipped copy in memory rather than the unzipped copy.

Reply:
Yeah, it's probably worth knowing both. That'll give us some better information to make a choice over later 🙂